spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -407,7 +508,12 @@ protected void whileLocked() throws IOException {
 						catch (IOException ex) {
 						}
 						try {
-							bos.close();
+							if (state == null || FileWritingMessageHandler.this.flushTask == null) {
```

Comment:
Good. Merging...
Although let me remove the `state` from the `fileStates` for the `stop()` case. To avoid unexpected effect on the next `start()`